diff_drive_controller/src/diff_drive_controller_parameter.yaml

```diff
@@ -39,6 +39,11 @@ diff_drive_controller:
     default_value: 1.0,
     description: "Correction factor when radius of right wheels differs from the nominal value in ``wheel_radius`` parameter.",
   }
+  tf_frame_prefix: {
+    type: string,
+    default_value: "",
+    description:  "Prefix to be appended to the tf frames, will be added to odom_id and base_frame_id before publishing",
```

Comment:
```suggestion
    description:  "(optional) Prefix to be appended to the tf frames, will be added to odom_id and base_frame_id before publishing. If the parameter is empty, controller's namespace will be used.",
```

Reply:
The main problem is that I don't want a namespace **at all** in my TF tree. 

I subscribe to the method of namespacing that the Nav2 package uses which remaps the whole TF topic to `/namespace/tf`. If by default it appends a namespace then it messes up the TF tree for those packages as suddenly I have a random `namespace/odom -> namespace/base_link` (Which also messes up styling as TF frame names shouldn't have `/` in them.
I think this method works best, because if you need a namespace appended to the frame name you can just add it in the params as a prefix when you launch the controller.

Sorry about the humble base, I'm a bit confused because humble and master are significantly different. #522 which is around the same sort of issue merged against humble also...

Reply:
@destogl Rebased to Master

Reply:
> Sorry about the humble base, I'm a bit confused because humble and master are significantly different. #522 which is around the same sort of issue merged against humble also...

No problem. Thanks for referencing this. It seems that I made a mistake by merging this into humble. This is not how it should be…

> I subscribe to the method of namespacing that the Nav2 package uses which remaps the whole TF topic to /namespace/tf. 

Sorry, I am not certain that I understand the issue completely. Do you have one or multiple robots in your system? If you have only one robot, then why use a controller under a specific namespace in the first place? Why don't you just use remapping of the `~/odometry_tf` topic? Would this work for you?

> If by default it appends a namespace then it messes up the TF tree for those packages as suddenly I have a random namespace/odom -> namespace/base_link (Which also messes up styling as TF frame names shouldn't have / in them.

I see your confusion here. This future targets multi-robot systems where automatic kind of prefixing is essential since you merge all `TF` messages into one topic or even is you use multiple ones, the visualization is very, very tricky to configure. This approach increases readability. As far as I understand, TF names should not have "/“ at the beginning of the name, but otherwise they are just strings. We decided to use "/" instead of "_" to increase readability when you are dealing with multiple robots. "_" is very often used in naming, therefore it is hard to see fast the difference between TF messages as when using "/" being more natural namespace delimiter.

Reply:
Yes I'm using multiple robots in my setup. 
In actual implementation of using a multi-robot system, having every robot publish into a single TF topic is bad practise due to communication issues that could arise (I'm dealing with 10-20 robots on WiFi here).
As such I treat every robot as an "island" with all the topics including `/tf` are in their own namespaces. It can only see itself and forwards messages via a broker, for debugging I relay parts of the TF tree such as `base link `and `odom` frames periodically with their namespace appended so that I can view the robot fleet as a whole. A 2 robot example is shown here:
![image](https://user-images.githubusercontent.com/94607409/221944970-ba9e66a7-8c6d-4cf5-b75d-e7048a65c7b5.png)
(View TF is a bit wide, but you get the idea...)

I get the argument between `/` and `_` and it doesn't really matter, apart from you end up with odd things such as having to make sure you remember to strip leading `/` as seen in #522 to adhere to the TF2 standard. 

What does matter is that I need to have the option of having a node in a namespace without appending the namespace to my TF frame names otherwise things get messy with the tree. Below shows what was happening in my setup with the current Humble implementation:
![image](https://user-images.githubusercontent.com/94607409/221941265-b46866aa-46e6-482a-b5fe-d4f05f01beed.png)
As you can see the tree is broken. 
My fix allows for there to be no `r0/` appended to the `odom` and `base_link` frames:
![image](https://user-images.githubusercontent.com/94607409/221942255-aa8a0357-cbde-42c8-868f-5b03a4052928.png)
Or if you want a prefix on the TF then you can do that too:
![image](https://user-images.githubusercontent.com/94607409/221942582-445a54ff-febf-4790-bfd8-7b9d504ecafe.png)

Reply:
@destogl I've just pushed a change that should address your comments. 

- I've switched back to `/` delimiters.
- Added a new tf_frame_prefix enable param that is true by default. 
- The tf_frame_prefix is blank by default. 
- And the default behaviour is to use the namespace if the tf_frame_prefix is blank.
This way I can turn off the prefix altogether and this solves my issue. 


I removed the `controller_namespace = controller_namespace.erase(0, 1) + "/";` from #522 in favour of a check to see if there's a single `/` as this seems to do the job and the erase was confusing.

Hope this resolves this!